cluster/addons/addon-manager/OWNERS

```diff
@@ -2,3 +2,7 @@
 
 reviewers:
 - mrhohn
+approvers:
+- mrhohn
+labels:
+- sig/cluster-lifecycle
```

Comment:
Is this used anywhere other than GCP? Does cluster-lifecycle actually want to own this?

Reply:
Good point, updated this to `sig/gcp`.